docs/latest/design/egctl.md

```diff
@@ -0,0 +1,61 @@
+# Introduce egctl
+
+## Motivation
+
+EG should provide a command line tool with following capabilities:
+
+- Collect configuration from envoy proxy and gateway
+- Update admin options such as changing log level
+- Analyse system configuration to diagnose any issues in envoy gateway
+
+This tool is named `egctl`.
+
+## Syntax
+
+Use the following syntax to run `egctl` commands from your terminal window:
+
+```console
+egctl [command] [entity] [name] [flags]
+```
+
+where `command`, `name`, and `flags` are:
+
+* `command`: Specifies the operation that you want to perform on one or more resources,
+  for example `config`, `version`.
+
+* `entity`: Specifies the entity the operation is being performed on such as `envoy-proxy` or `envoy-gateway`.
+
+* `name`: Specifies the name of the specified instance. 
+
+* `flags`: Specifies optional flags. For example, you can use the `-c` or `--config` flags to specify the values for installing.
+
+If you need help, run `egctl help` from the terminal window.
+
+## Operation
+
+The following table includes short descriptions and the general syntax for all the `egctl` operations:
+
+| Operation | Syntax                           | Description                                                                 |
+| --------- | -------------------------------- | --------------------------------------------------------------------------- |
+| `version` | `egctl version`                  | Prints out build version information.                                       |
+| `config`  | `egctl config ENTITY`            | Retrieve information about proxy configuration from envoy proxy and gateway |
+| `log`     | `egctl log ENTITY --level trace` | Change envoy proxy's log level                                              |
+| `analyze` | `egctl analyze`                  | Analyze EG configuration and print validation messages                      |
```

Comment:
Can you help me better understand the `analyze` subcommand. It might be better to remove this subcommand for now and have egctl focus on `proxy-config` to retrieve the proxy bootstrap|cluster|route|etc configs and `version`.

Reply:
this is inspired from istio where `analyze` is used to collect information from multiple sources and surface any runtime issues to the end user
https://istio.io/latest/docs/ops/diagnostic-tools/istioctl-analyze/